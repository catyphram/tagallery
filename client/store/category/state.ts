export interface Category {
  id: string
  name: string
  description?: string
}

const state = () => ({
  data: [] as Category[],
  error: undefined,
  loading: false,
})

export type State = ReturnType<typeof state>

export default state
