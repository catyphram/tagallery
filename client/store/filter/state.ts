import type { Category } from '../category/state'

export enum Status {
  Processed,
  Unprocessed,
}
export enum Mode {
  View,
  Verify,
}
export enum Categorization {
  Done,
  Open,
}

const state = () => ({
  status: Status.Processed,
  mode: Mode.View,
  categorization: Categorization.Done,
  categories: [] as Category[],
})

export type State = ReturnType<typeof state>

export default state
