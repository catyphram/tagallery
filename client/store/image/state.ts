export interface Image {
  file: string
  assignedCategories: string[]
  proposedCategories: string[]
  starredCategory?: string
}

const state = () => ({
  data: [] as string[],
  error: undefined,
  loading: false,
})

export type State = ReturnType<typeof state>

export default state
