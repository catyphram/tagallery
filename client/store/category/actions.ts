import { ActionTree } from 'vuex'

import { Category, State } from './state'

const mockCategories = [
  {
    id: 'category1',
    name: 'Category 1',
    description: 'Category 1 Description',
  },
  {
    id: 'category2',
    name: 'Category 2',
    description: 'Category 2 Description',
  },
  {
    id: 'category3',
    name: 'Category 3',
    description: 'Category 3 Description',
  },
  {
    id: 'category4',
    name: 'Category 4',
    description: 'Category 4 Description',
  },
  {
    id: 'category5',
    name: 'Category 5',
    description: 'Category 5 Description',
  },
]

const actions: ActionTree<State, State> = {
  load({ commit }) {
    commit('load')
    commit('receive', {
      data: mockCategories,
    })
  },
  update({ commit }, { category }: { category: Category }) {
    commit('update', { category })
  },
  add({ commit }, { category }: { category: Category }) {
    commit('add', { category })
  },
}

export default actions
