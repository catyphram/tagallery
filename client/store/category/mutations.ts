import { MutationTree } from 'vuex'

import { Category, State } from './state'

const mutations: MutationTree<State> = {
  load(state: State) {
    state.loading = true
  },
  receive(state: State, { data = [] as Category[], error = undefined } = {}) {
    state.loading = false
    state.data = data
    state.error = error
  },
  update(state: State, { category }: { category: Category }) {
    const stateCategory = state.data.find(
      (stateCategory) => stateCategory.id === category.id
    )
    if (stateCategory) {
      stateCategory.name = category.name
      stateCategory.description = category.description
    }
  },
  add(state: State, { category }: { category: Category }) {
    state.data.push(category)
  },
}

export default mutations
