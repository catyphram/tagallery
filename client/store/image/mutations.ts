import { MutationTree } from 'vuex'

import { Image, State } from './state'

const mutations: MutationTree<State> = {
  load: (state: State) => {
    state.loading = true
  },
  receive: (state: State, { data = [] as Image[], error = undefined } = {}) => {
    state.loading = false
    state.data = data
    state.error = error
  },
}

export default mutations
