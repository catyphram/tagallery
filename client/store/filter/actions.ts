import { ActionTree } from 'vuex'
import { Category } from '../category/state'

import { Categorization, Mode, State, Status } from './state'

const actions: ActionTree<State, State> = {
  set(
    { commit },
    {
      status,
      mode,
      categorization,
      categories,
    }: {
      status: Status
      mode: Mode
      categorization: Categorization
      categories: Category[]
    }
  ) {
    commit('set', {
      status,
      mode,
      categorization,
      categories,
    })
  },
}

export default actions
