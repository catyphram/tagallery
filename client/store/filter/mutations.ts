import { MutationTree } from 'vuex'
import { Category } from '../category/state'

import { Categorization, Mode, State, Status } from './state'

const mutations: MutationTree<State> = {
  set(
    state: State,
    {
      status,
      mode,
      categorization,
      categories,
    }: {
      status?: Status
      mode?: Mode
      categorization?: Categorization
      categories?: Category[]
    }
  ) {
    state.status = status ?? state.status
    state.mode = mode ?? state.mode
    state.categorization = categorization ?? state.categorization
    state.categories = categories ?? state.categories
  },
}

export default mutations
