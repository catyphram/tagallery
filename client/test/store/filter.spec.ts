import { Category } from '~/store/category/state'
import mutations from '~/store/filter/mutations'
import { Categorization, Mode, State, Status } from '~/store/filter/state'

const getDefaultState = (): State => ({
  status: Status.Processed,
  mode: Mode.View,
  categorization: Categorization.Done,
  categories: [] as Category[],
})

describe('filter', () => {
  describe('mutation', () => {
    describe('set', () => {
      test('should set new options', () => {
        const state = getDefaultState()
        const updates = {
          status: Status.Unprocessed,
          mode: Mode.Verify,
          categorization: Categorization.Open,
          categories: [
            {
              id: 'cat1',
              name: 'Category 1',
              description: 'Category 1 Description',
            },
          ],
        }

        mutations.set(state, updates)

        expect(state).toMatchSnapshot()
      })
    })
  })
})
