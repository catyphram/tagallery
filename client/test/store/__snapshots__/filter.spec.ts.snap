// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter mutation set should set new options 1`] = `
Object {
  "categories": Array [
    Object {
      "description": "Category 1 Description",
      "id": "cat1",
      "name": "Category 1",
    },
  ],
  "categorization": 1,
  "mode": 1,
  "status": 1,
}
`;
